README.md

```diff
@@ -355,6 +356,38 @@ writable.
 
 Always prefer `ReadonlyArray` over a regular `Array` unless it must be possible to modify the Array.
 
+### React props & TypeScript
+
+`interface` is preferred over `type` when defining the component props. Are a little more powerful
+than the corresponding type declaration, but for a set of react props, it likely doesn't matter.
```

Comment:
For the standards, I think we should never have a `it likely doesn't matter` case as it creates ambiguity over the standard. When exactly it would and wouldn't matter?

Regarding `interface` vs `type`, I think over the past year there has been a shift back to `interface` over the performance concerns of `type`. It is also noted by the TS team [here](https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections).

I think that we should just decide as a team which one to use over the other. I personally would be fine switching to `interface` and using `type` only when I absolutely need to.

Reply:
Could you check again?

Reply:
Using an `interface` when you don't use it's semantics is wrong IMO, if you're not going to declaration merge it, it should not be able to. Furthermore there are a few edge case issues with mapped interfaces surrounding (non-)indexed keys that can be cumbersome (or from lib impossible) to overcome.

I prefer types unless you need to be able to declaration merge (almost never for non-library code).

Reply:
@mmnathan 
I think we agree about using types, but it would be nice to have a section documenting the reason for using types over interfaces. Also explaining where interfaces could still be useful, https://github.com/mediamonks/frontend-coding-standards/issues/93